install.sh

```diff
@@ -8,33 +10,75 @@ print_usage() {
     echo "  -h    Show this help message"
 }
 
+retry() {
+    # This facilitates arbitrary retry logic for the provided command
+    #
+    # USAGE:
+    #  # Use the defaults
+    #  retry curl --fail https://example.com
+    #  # Provide an updated count
+    #  retry 10 curl --fail https://example.com
+    #  # Provide an updated count and delay
+    #  retry 5 10 curl --fail https://example.com
+    local retry_count=3
+    local retry_delay=2
+
+    # Update the default values if provided
+    [[ $1 =~ ^[0-9]+$ ]] && { retry_count=$1; shift; }
+    [[ $1 =~ ^[0-9]+$ ]] && { retry_delay=$1; shift; }
+
+    local n=1
+    until "$@"; do
+        if (( n >= retry_count )); then
+            printf '⚠️  attempt %d/%d failed for: %s\n' "$n" "$retry_count" "$*" >&2
+            return 1
+        fi
+        printf '⚠️  attempt %d/%d failed for: %s; retrying in %ds\n' "$n" "$retry_count" "$*" "$retry_delay" >&2
+        sleep "$retry_delay"
+        ((n++))
+    done
+}
+
+
 # Function to get available versions - already checked when running main
 get_available_versions() {
     command -v curl >/dev/null 2>&1 || { echo >&2 "Required tool curl could not be found. Aborting."; exit 1; }
-    curl -s https://api.github.com/repos/opengrep/opengrep/releases | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/'
+    # This extracts and validates the version tag(s) without requiring jq
```

Comment:
But there was no use of `jq`, not sure if this comment is needed?

Reply:
Removed in ee1923ddde7ecb755f2dba25fe8d65509dd5284e